src/components/block/index.ts

```diff
@@ -505,14 +446,21 @@ export default class Block extends EventsDispatcher<BlockEvents> {
    * @param {boolean} state - 'true' to select, 'false' to remove selection
    */
   public set selected(state: boolean) {
-    if (state) {
-      this.holder.classList.add(Block.CSS.selected);
+    this.holder.classList.toggle(Block.CSS.selected, state);
 
-      SelectionUtils.addFakeCursor(this.holder);
-    } else {
-      this.holder.classList.remove(Block.CSS.selected);
+    const fakeCursorWillBeAdded = state === true && SelectionUtils.isRangeInsideContainer(this.holder);
+    const fakeCursorWillBeRemoved = state === false && SelectionUtils.isElementContainsFakeCursor(this.holder);
+
+    if (fakeCursorWillBeAdded || fakeCursorWillBeRemoved) {
+      this.editorEventBus?.emit(FakeCursorAboutToBeSet, { state }); // mutex
```

Comment:
Why those constants are not enum?

Reply:
Also why event is called `AboutToBeSet` even if cursor will be removed

Reply:
renamed to `FakeCursorAboutToBeToggled`. 

Enum is not suitable here, because all the information about the single event stored in isolated file. There is no need to describe a enum with all available event names — EventMap used instead

<img width="823" alt="image" src="https://github.com/codex-team/editor.js/assets/3684889/d6ecd9df-b289-47d2-8dd8-13c3574060bf">